File: ingestWorker/internal/storage/entity/ingestJobEntity.go
package entity

import "time"

type IngestJobEntity struct {
	tableName struct{} `pg:"ingest_jobs"`

	ID           string     `pg:"ingest_job_id,pk"`
	SubmissionID string     `pg:"submission_id,notnull"`
	Status       string     `pg:"status,notnull"`
	CreatedAt    time.Time  `pg:"created_at,notnull"`
	StartedAt    *time.Time `pg:"started_at"`
	FinishedAt   *time.Time `pg:"finished_at"`
	ErrorMessage *string    `pg:"error_message"`
}

File: ApiService/Infrastructure/Entities/IngestJobEntity.cs
namespace Infrastructure.Entities;

public class IngestJobEntity
{
    public Guid Id { get; set; }

    public Guid SubmissionId { get; set; }

    public string Status { get; set; } = null!;

    public DateTimeOffset CreatedAt { get; set; }

    public DateTimeOffset? FinishedAt { get; set; }

    public string? ErrorMessage { get; set; }
}
